src/bio/bio_xstream.c

```diff
@@ -400,7 +417,7 @@ bio_nvme_init(const char *nvme_conf, int shm_id, int mem_size,
 	nvme_glb.bd_bs_opts.max_channel_ops = BIO_BS_MAX_CHANNEL_OPS;
 
 	bio_chk_cnt_init = DAOS_DMA_CHUNK_CNT_INIT;
-	bio_chk_cnt_max = DAOS_DMA_CHUNK_CNT_MAX;
+	bio_chk_cnt_max = (mem_size / tgt_nr) / size_mb;
```

Comment:
As we discussed, the upper bound should be smaller than the (total configured size / target count), otherwise, allocation could fail when used pages closing to upper bound.

for example, set it bio_chk_cnt_max -= (200 / size_mb)? I'm not sure if 200MB buffer between upper bound and configured limit is a proper value.

Reply:
@NiuYawei right, let me do some more investigation on this buffer value

Reply:
After some testing and discussion with @NiuYawei , we will be setting the buffer to be 200 hugepages per target (~400MB buffer with 2MB hugepage size). There may be some outlier cases where allocation may still fail (requires more than 200 hugepages), and we plan to handle these allocation failures in a separate patch.